src/slurm_plugin/instance_manager.py

```diff
@@ -1008,9 +1008,13 @@ def _launch_instances(
                 if slurm_node_list:
                     logger.info(
                         "Launching %s instances for nodes %s",
-                        "all-or-nothing" if all_or_nothing_batch else "best-effort",
+                        scaling_strategy,
                         print_with_count(slurm_node_list),
                     )
+                    # At instance launch level, the various scaling strategies can be grouped based on the actual
+                    # launch behaviour i.e. all-or-nothing or best-effort
+                    all_or_nothing_batch = scaling_strategy in [ScalingStrategy.ALL_OR_NOTHING]
+
```

Comment:
In `all_or_nothing_batch` the `_batch` alludes to the way we chunk the launch attempts in "batches" of a certain size/count of instances. So `all_or_nothing_batch` implies that each batch will be done in an all-or-nothing fashion.

> as done a few lines above, I assume that this is because we are planning to add another strategy that falls in the "all or nothing" category.

Exactly, it was done with this in mind.